packages/bot/src/scratch/hooks/toolbox.js

```diff
@@ -80,122 +80,135 @@ Blockly.Toolbox.prototype.populate_ = function (newTree) {
  * @private
  */
 Blockly.Toolbox.prototype.showCategory_ = function (category_id) {
-    let flyout_content;
+    let flyout_content = [];
+    let is_search = false;
 
     if (category_id === 'search') {
-        let search_term = document.getElementById('search_input').value;
-        const all_variables = this.flyout_.workspace_.getVariablesOfType('');
+        const search_term = document.getElementById('search_input').value.toUpperCase();
+        const all_variables = Blockly.derivWorkspace.getVariablesOfType('');
         const all_procedures = Blockly.Procedures.allProcedures(Blockly.derivWorkspace);
```

Comment:
Better to define both `all_variables` and `all_procedures` variables close to their concerns, the first time these variables are being used is 100 lines later.